packages/kbn-unified-data-table/src/hooks/use_full_screen_watcher.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useGeneratedHtmlId, useMutationObserver } from '@elastic/eui';
+import { useCallback, useState } from 'react';
+
+export const useFullScreenWatcher = () => {
+  const dataGridId = useGeneratedHtmlId({ prefix: 'unifiedDataTable' });
+  const [dataGridWrapper, setDataGridWrapper] = useState<HTMLElement | null>(null);
+  const [dataGrid, setDataGrid] = useState<HTMLElement | null>(null);
+
+  const checkForDataGrid = useCallback<MutationCallback>(
+    (_, observer) => {
+      const foundDataGrid = document.getElementById(dataGridId);
+
+      if (foundDataGrid) {
+        setDataGrid(foundDataGrid);
+        observer.disconnect();
+      }
+    },
+    [dataGridId]
+  );
```

Comment:
The `ref` from `EuiDataGrid` doesn't provide access to the data grid DOM element, so we need to use a mutation observer and access it by ID.